Modules/Loadable/Segmentations/EditorEffects/Python/SegmentEditorHollowEffect.py

```diff
@@ -154,49 +134,29 @@ def onApply(self):
     thresh.SetInValue(backgroundValue)
     thresh.SetOutValue(labelValue)
     thresh.SetOutputScalarType(selectedSegmentLabelmap.GetScalarType())
-    thresh.Update()
 
-    subtract = vtk.vtkImageMathematics()
-    subtract.SetOperationToSubtract()
-    kernelSizePixel = self.getKernelSizePixel()
     shellMode = self.scriptedEffect.parameter("ShellMode")
+    shellThicknessMm = self.scriptedEffect.doubleParameter("ShellThicknessMm")
+    import vtkITK
+    margin = vtkITK.vtkITKImageMargin()
+    margin.SetInputConnection(thresh.GetOutputPort())
     if shellMode == MEDIAL_SURFACE:
-      dilateKernelSizePixel = [int((kernelSizePixel[i]+1.0)/4.0)*2+1 for i in range(3)]
-      erodeKernelSizePixel = [int((kernelSizePixel[i]-1.0)/4.0)*2+1 for i in range(3)]
+      margin.SetOuterMarginMm(0.5*shellThicknessMm)
+      margin.SetInnerMarginMm(-0.5*shellThicknessMm)
     elif shellMode == INSIDE_SURFACE:
-      dilateKernelSizePixel = kernelSizePixel
-      erodeKernelSizePixel = [1, 1, 1]
+      margin.SetOuterMarginMm(shellThicknessMm)
+      margin.SetInnerMarginMm(0.0 + 0.000001)
```

Comment:
We don't know what the size of a voxel is in mm (it might be smaller than 0.000001). It would be safer to use 0.001*voxelDiameter.